### DISCUSSION TITLE
IDE friendly development

### DISCUSSION THREAD

**Discussion by prange** [dillonkearns/elm-pages#247] [Q&A] [answered]
I am not sure where to place this question, please feel free to move it to an appropriate location :)

I have just fired up elm-pages and started to take a look with the Elm IDE plugin to Intellij. Is there a way to tell the IDE where the Main.elm file is so it can compile the source? This is really useful for code-completion etc.

**Comment by dillonkearns** [comment, marked as answer]
Hello @prange, I use intellij-elm as my main IDE as well and it works with elm-pages. Be sure that you've run either `elm-pages build` or `elm-pages dev` and opened one of the pages first, otherwise it may not have everything setup properly.

Once you do that, you'll need to manually point Intellij to your `elm.json` file for the project. Unfortunately that's just the workflow for any Elm project with intellij-elm (see https://github.com/klazuka/intellij-elm/issues/583). Hope that helps!
